### ISSUE TITLE
Benchmarking YOLO untuk Autonomous Underwater Vehicle (AUV)

### ISSUE THREAD

**Issue by ezra1702** [ezra1702/Transfer-Learning-Amarine-Vision#29]
# Mini Project Break Semester :  Benchmarking YOLO untuk Autonomous Underwater Vehicle (AUV)

## 1. Overview

Tim **Amarine Vision** sedang bekerja sama dengan **Lembaga Badan Riset dan Inovasi Nasional (BRIN)** mengembangkan sistem Computer Vision untuk **Autonomous Underwater Vehicle (AUV)** robot bawah air yang membantu monitoring ekosistem laut: mendeteksi ikan, terumbu karang, sampah laut, ubur-ubur, dan objek bawah laut lainnya.

Sebagai anggota **Visioneer**, kalian ditugaskan membangun dataset bawah laut, melatih **minimal 3 model YOLO**, dan melakukan benchmarking untuk merekomendasikan model terbaik bagi prototipe AUV Amarine Vision. Fokus penilaian ada pada proses eksplorasi dan analisis, bukan semata angka akurasi tertinggi.

---

## 2. Misi Utama

> Membangun dataset deteksi objek bawah laut, melatih **minimal 3 model YOLO** berbeda, lalu melakukan benchmarking untuk merekomendasikan model terbaik bagi prototipe AUV — dengan mempertimbangkan akurasi, kecepatan, dan efisiensi komputasi.

---

### 3.1 Dataset
Tema **wajib underwater**, objek bebas (contoh: fish, coral, jellyfish, sea turtle, starfish, crab, shrimp, seaweed, plastic waste, bottle, fishing net, anchor, dll).

**Minimum:** 3 kelas, 20 gambar unik/kelas, total ≥60 gambar. Sumber bebas: Google Images, Kaggle, Roboflow Universe, Open Images, foto pribadi, dataset publik lain.

Variasi data (sudut, cahaya, kejernihan air, oklusi) penting agar model tidak overfitting dan mampu generalisasi ke kondisi nyata.

### 3.2 Annotation (pakai Roboflow)
Create Project → Upload gambar → Bounding Box tiap objek → Quality Checking (review ulang label & box). Anotasi buruk = performa model buruk, walau arsitektur bagus. (kalau kesulitan bisa tanyak atau nonton tutorial youtube :  https://www.youtube.com/watch?v=abqKjf9O-as 

### 3.3 Dataset Preparation
Di Roboflow, minimal:
- **Preprocessing:** Auto Orient, Resize 640x640
- **Augmentation (pilih ≥3):** Horizontal Flip, Rotation, Brightness, Blur, Noise, Contrast, Saturation, Exposure

Augmentasi penting karena kondisi bawah laut sangat bervariasi (cahaya, warna air, blur akibat partikel) — augmentasi mensimulasikan variasi ini tanpa perlu ribuan foto asli.

### 3.4 Export Dataset
Format **YOLO**. Struktur folder:
```
underwater-dataset/
├── train/{images,labels}/
├── valid/{images,labels}/
├── test/{images,labels}/
└── data.yaml
```

### 3.5 Training
**Wajib minimal 3 model YOLO** (contoh: YOLOv5, YOLOv8, YOLO11, atau lainnya — bebas, sesuaikan spesifikasi laptop). Parameter bebas (epoch, batch size, learning rate, optimizer, image size, workers, confidence threshold, IoU threshold), **wajib dicantumkan di README**.

---

## 4. Benchmarking

Bandingkan minimal: **Precision, Recall, mAP50, mAP50-95, Training Time, Inference Time, FPS, Ukuran Model, Jumlah Parameter** (jika memungkinkan). Opsional: FLOPs, GPU Memory, CPU Usage.

Contoh format tabel:

| Model | Precision | Recall | mAP50 | mAP50-95 | Training Time | Inference Time | FPS | Ukuran Model | Params |
|---|---|---|---|---|---|---|---|---|---|
| YOLOv5s | 0.87 | 0.81 | 0.85 | 0.62 | 42 menit | 12 ms | 83 | 14.1 MB | 7.2 M |
| YOLOv8n | 0.89 | 0.78 | 0.86 | 0.60 | 35 menit | 8 ms | 125 | 6.3 MB | 3.2 M |
| YOLO11s | 0.91 | 0.84 | 0.89 | 0.66 | 50 menit | 15 ms | 66 | 18.4 MB | 9.4 M |

*(angka di atas hanya contoh format, bukan hasil sungguhan)*. Sertakan juga visualisasi (bar/radar chart) agar perbandingan mudah dibaca.

---

## 5. Inference

Uji minimal **10 gambar** atau **1 video ≥15 detik**. Bandingkan hasil ketiga model: objek yang terlewat (false negative), deteksi palsu (false positive), confidence score, dan kecepatan visual. Sertakan screenshot hasil deteksi tiap model.

---

## 6. Analisis

Wajib dijawab dengan alasan teknis, bukan sekadar menyebut angka:
1. Mengapa Model A lebih akurat?
2. Mengapa Model B lebih cepat?
3. Apa trade-off yang terlihat dari hasil kalian?
4. Model mana paling cocok untuk AUV secara keseluruhan?
5. Model mana yang dipilih jika hardware robot sangat terbatas?

---

## 7. Eksplorasi Tambahan

Wajib **minimal 1 eksperimen tambahan**, contoh: ubah image size, epoch, batch size, augmentasi, confidence/IoU threshold, optimizer, coba model YOLO lain, atau tambah dataset. Semua perubahan **wajib dianalisis** (bandingkan sebelum-sesudah).

---

## 8. Struktur Repository
```
underwater-yolo-benchmark-[Nama]/
├── README.md
├── requirements.txt
├── dataset/{train,valid,test}/ + data.yaml
├── models/{yolov5s,yolov8n,yolo11s}/best.pt
├── notebooks/  (data prep, training tiap model, benchmarking)
├── results/  (tabel & chart benchmark, confusion matrix, training curves)
├── inference/{input,output}/
└── docs/additional_experiment.md
```

---

## 9. README
Wajib memuat:  deskripsi dataset & link Roboflow, detail augmentasi, daftar model & parameter training, hasil benchmarking, hasil inference, analisis (Bagian 7), eksperimen tambahan (Bagian 8), kesimpulan & rekomendasi, cara reproduksi project, referensi.

---
## 10. Rubrik Penilaian

| No | Komponen | Bobot |
|---|---|---|
| 1 | Kualitas & variasi dataset | 15% |
| 2 | Annotation & dataset preparation (Roboflow) | 10% |
| 3 | Training minimal 3 model YOLO | 20% |
| 4 | Kelengkapan benchmarking | 20% |
| 5 | Hasil & perbandingan inference | 10% |
| 6 | Kedalaman analisis | 15% |
| 7 | Eksplorasi tambahan | 10% |
| **Total** | | **100%** |

---

## 11. Deadline

📅 **Sabtu, 31 Agustus 2026, 23.59 WIB**

---
